### PR TITLE
add missing const qualifiers

### PR DESCRIPTION
**Pull request by kroening**
This prevents accidental disruption of sharing.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- n/a My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 032fd49a049411f2ed11e44da12d977bf78b42ab).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101180978
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: a6df10fe42dd388506de5b79a4d10d56275aed0f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101181145
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 78725e3e8fce8021b35964abbfa3b9ed37e088df).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101181415

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: a83a5fc20c7e0cc0244752d704971f1787f026da).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101187394

File: src/goto-programs/string_abstraction.cpp [hunk 1]
```diff
@@ -1046,13 +1046,13 @@ void string_abstractiont::move_lhs_arithmetic(exprt &lhs, exprt &rhs)
 
 goto_programt::targett string_abstractiont::abstract_pointer_assign(
   goto_programt &dest,
-  goto_programt::targett target)
+  const goto_programt::targett target)
```
Comment (tautschnig) [inline src/goto-programs/string_abstraction.cpp:1049]: See other PR: I'm not sure this adds value.

**Review by tautschnig — APPROVED** [review]

File: src/goto-programs/string_abstraction.cpp (hunk 1 above)
Comment (kroening) [reply src/goto-programs/string_abstraction.cpp:1049]: Well, these do add exactly as much value as on block-scoped variables, and we have a tendency to make these `const` unless they get mutated. There's of course more value on references.

Comment (kroening) [reply src/goto-programs/string_abstraction.cpp:1049]: As a proposal, I'd say we add the `const` on the definition, but leave it out of the declaration (where it really doesn't do anything at all).

**Review by smowton — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**